From 675517a715ee74ecd72ce359420bfa593c3fac57 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Wed, 7 Jan 2026 14:15:40 +0100
Subject: [PATCH] arm64: dts: qcom: ipq6018: fix WCSS node

WCSS has never worked with the current node as there is no actual driver
support for the secure WCSS that is found inside of IPQ6018.

Now that there is a dedicated WCSS secure PIL driver, use that to fix
WCSS support.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 arch/arm64/boot/dts/qcom/ipq6018.dtsi | 28 +++++++++------------------
 1 file changed, 9 insertions(+), 19 deletions(-)

--- a/arch/arm64/boot/dts/qcom/ipq6018.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq6018.dtsi
@@ -835,11 +835,10 @@
 		};
 
 		q6v5_wcss: remoteproc@cd00000 {
-			compatible = "qcom,ipq6018-wcss-pil";
-			reg = <0x0 0x0cd00000 0x0 0x4040>,
-			      <0x0 0x004ab000 0x0 0x20>;
-			reg-names = "qdsp6",
-				    "rmb";
+			compatible = "qcom,ipq6018-wcss-sec-pil", "qcom,ipq9574-wcss-sec-pil";
+			reg = <0x0 0x0cd00000 0x0 0x4040>;
+			firmware-name = "IPQ6018/q6_fw.mdt",
+					"IPQ6018/m3_fw.mdt";
 			interrupts-extended = <&intc GIC_SPI 325 IRQ_TYPE_EDGE_RISING>,
 					      <&wcss_smp2p_in 0 0>,
 					      <&wcss_smp2p_in 1 0>,
@@ -851,24 +850,15 @@
 					  "handover",
 					  "stop-ack";
 
-			resets = <&gcc GCC_WCSSAON_RESET>,
-				 <&gcc GCC_WCSS_BCR>,
-				 <&gcc GCC_WCSS_Q6_BCR>;
-
-			reset-names = "wcss_aon_reset",
-				      "wcss_reset",
-				      "wcss_q6_reset";
-
-			clocks = <&gcc GCC_PRNG_AHB_CLK>;
-			clock-names = "prng";
-
-			qcom,halt-regs = <&tcsr 0x18000 0x1b000 0xe000>;
+			clocks = <&gcc GCC_PRNG_AHB_CLK>,
+				 <&gcc GCC_QDSS_AT_CLK>;
+			clock-names = "prng",
+				      "qdss";
 
 			qcom,smem-states = <&wcss_smp2p_out 0>,
 					   <&wcss_smp2p_out 1>;
 			qcom,smem-state-names = "shutdown",
 						"stop";
-
 			memory-region = <&q6_region>;
 
 			glink-edge {
@@ -877,7 +867,7 @@
 				qcom,remote-pid = <1>;
 				mboxes = <&apcs_glb 8>;
 
-				qrtr_requests {
+				rpm_requests {
 					qcom,glink-channels = "IPCRTR";
 				};
 			};
